integrate/romberg.go

```diff
@@ -0,0 +1,98 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+)
+
+//
+// Helper method for getting the base 2 exponent of an integer
+//
+func getBase2Exponent(x int) int {
+	n := 1
+	k := 0
+
+	for n < x {
+		n <<= 1
+		k += 1
+	}
+	return k
+}
+
+// Romberg estimates the integral of the function f
+// \int_0^1 f(x)dx
+// The function f must be sampled at equal intervals.
+// Romberg's method approximates the integral of the function by
+// applying Richardson extrapolation. Hence, the function adaptively
+// increase the number of evaluation points until the desired accuracy
+// is obtained or the maximum number of evaluation points has been used.
+// See https://en.wikipedia.org/wiki/Romberg%27s_method for a thorough description
+// of the algorithm.
+//
+// The length of the length of the f array must be 2**k + 1, where k is a positive integer
+// The minimum length of the work array is 2*k
+// The method returns the value of the integral, and the number of levels needed to reach
+// the desired accuracy
+func Romberg(f []float64, acc float64, work []float64) (float64, int) {
+	ninterv := len(f) - 1
+	// Check that number of intervals is equal to a power of 2
```

Comment:
This is declared too early and the comment looks well out of place. Lower?

Reply:
This comment is removed